raster-foundry/raster-foundry
# Add task api routes to project layers

## Overview

This PR adds task routes described in the spec to project layers.

### Checklist


### Notes

I made a few executive decisions about the spec. The uncontroversial one is that deleting returns a 204 instead of a 201. The more controversial one might be https://github.com/raster-foundry/raster-foundry/compare/feature/js/add-task-endpoints?expand=1#diff-a5b6728a00e883410ca6406be9f5de0bL3479. Separate endpoints for creating and deleting locks felt a bit heavy when we already have verbs for creating and deleting the same thing. Also it meant I didn't have to spend as much time adding sets of curly braces and making sure that my thing that compiles actually does routing correctly (thanks akka-http).

## Testing Instructions

- run tests
- check structure from [project routes](https://github.com/raster-foundry/raster-foundry/blob/feature/js/add-task-endpoints/app-backend/api/src/main/scala/project/Routes.scala#L290-L320) matches the spec
- confirm authorization rules make sense in: https://github.com/raster-foundry/raster-foundry/blob/feature/js/add-task-endpoints/app-backend/api/src/main/scala/project/ProjectLayerTaskRoutes.scala
- make sure tests pass
 
Closes #4983
